### PR TITLE
Detail view design

### LINKED ISSUE #18: 16. As a user, I want the item detail screen to look pretty
AC:

* The item detail screen (the screen a user sees when they tap on one of their items in the shopping list) looks something like the following (or equivalent, use your creativity!):

item-detail

### PR DESCRIPTION
**Pull request by skillitzimberg**
## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|     | :sparkles: New feature     |
| ✓   | :sunglasses: Styling       |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Related Issue

closes #18 

## Description

The UI from clicking a button to view an item's details to the display of that item's details is cohesive with the rest of the app's styling.

We incorporated a modal view for the item details.

## List To Complete Task

- [x] Fix bugs in the modal display and routing.
- [x] Add color theme to header/footer.
- [x] Style the buttons appropriately.

### Before
WITHOUT MODAL:
Screen Shot 2020-05-25 at 12 30 53 PM

### After
Screen Shot 2020-05-26 at 11 54 32 AM

## Acceptance Criteria

- The item detail view is aligned with the app's theme.

## Testing Steps / QA Criteria

View the deployed branch [here](https://deploy-preview-48--tcl-6-smart-shopping-list.netlify.app/).
Join a shopping list (`stamp avow boil` is a valid token) or view your own shopping list.
Click <some HTML element> (As of my starting this PR it is the "View Details" link on an item).
A modal with the relevant details for the item is displayed and looks good.

### CODE REVIEW

File: src/components/ItemDetail.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import { Link, useParams } from 'react-router-dom';
 import arrowLeft from '../image/arrow-left.svg';
 import '../CSS/Icon.css';
-
+// Small change
```
Comment (laurenmbeatty) [inline src/components/ItemDetail.js]: 😂

Comment (laurenmbeatty) [reply src/components/ItemDetail.js]: Maybe we can get rid of this now.  : )

**Comment by laurenmbeatty** [comment]
This looks great. I'm wondering about this:

Right now, if I have the modal showing, and the id is in the url, and then refresh the browser, the url stays the same, but the modal is no longer showing. I wonder if there is a way to have that persist?

File: src/CSS/DetailModal.css [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+  padding: 1rem 1rem;
+  border-bottom: 1px solid #dee2e6;
+  border-top-left-radius: calc(0.3rem - 1px);
+  border-top-right-radius: calc(0.3rem - 1px);
+}
+
+.modal-header .close {
+  padding: 1rem 1rem;
+  margin: -1rem -1rem -1rem auto;
+}
```
Comment (laurenmbeatty) [inline src/CSS/DetailModal.css]: Is this needed anymore? It doesn't look like there is a close button in the modal header now.

**Comment by SaraSweetie** [comment]
Since the back arrow is no longer used it can be removed and any reference to it. Also, the CSS color variables were copied into this file, and that's unnecessary and duplicating them from App.css

**Comment by skillitzimberg** [comment]
> Since the back arrow is no longer used it can be removed and any reference to it. Also, the CSS color variables were copied into this file, and that's unnecessary and duplicating them from App.css

Back arrow has been removed.

The Shopping list branch has a `colors.css` file comprising all the color variables. Color variables have been removed from all other css files. I think we can rely on the fix there once everything is merged together. Otherwise we might end up with more merge conflicts because we'll be changing the same lines across branches.

**Comment by skillitzimberg** [comment]
> This looks great. I'm wondering about this:
> 
> Right now, if I have the modal showing, and the id is in the url, and then refresh the browser, the url stays the same, but the modal is no longer showing. I wonder if there is a way to have that persist?

I think I fixed this.

File: src/components/DetailModal.js [hunk 3]
```diff
@@ -1,17 +1,22 @@
 import React from 'react';
+import { useParams } from 'react-router-dom';
 import Modal from 'react-bootstrap/Modal';
 import Button from 'react-bootstrap/Button';
 
 import '../CSS/Icon.css';
 
 const DetailModal = ({ show, handleClose, item }) => {
+  let showModal = show;
+  const url = useParams();
```
Comment (laurenmbeatty) [inline src/components/DetailModal.js:10]: nice!

File: src/components/ListItem.js [hunk 4]
```diff
@@ -31,7 +36,7 @@ const ListItem = ({ item, onDelete, token }) => {
     setPurchased(true);
   }
 
-  function onHandle(event) {
+  function onPurchase(event) {
```
Comment (laurenmbeatty) [inline src/components/ListItem.js:39]: good name.

**Comment by laurenmbeatty** [comment]
Awesome, routing really works now. Good job. There is another way to approach it, (I think, anyway), where the useParams could live in the ListItem file. That way, you'd only be passing `show` to the DetailsModal, and not having to check inside of the Modal itself whether or not it should show. However, not a blocker, just another way to approach it. Well done.

**Review by laurenmbeatty — APPROVED** [review]
LGTM 👍

**Review by SaraSweetie — APPROVED** [review]

**Pull request merged**